openbook_auth/models.py

```diff
@@ -1445,23 +1445,23 @@ def get_community_post_with_id(self, post_id):
 
         post_query.add(~Q(community__banned_users__id=self.pk), Q.AND)
 
+        # Check we have membership or its a public community
         post_query_visibility_query = Q(community__memberships__user__id=self.pk)
         post_query_visibility_query.add(Q(community__type=Community.COMMUNITY_TYPE_PUBLIC, ), Q.OR)
-        post_query_visibility_query = Q(community__memberships__user__id=self.pk, is_closed=False)
 
-        post_query_visibility_query.add(Q(community__memberships__user__id=self.pk,
-                                          community__memberships__is_administrator=True,
-                                          is_closed=True), Q.OR)
-        post_query_visibility_query.add(Q(community__memberships__user__id=self.pk,
-                                          community__memberships__is_moderator=True,
-                                          is_closed=True), Q.OR)
-        post_query_visibility_query.add(Q(community__memberships__user__id=self.pk,
-                                          creator=self,
-                                          is_closed=True), Q.OR)
+        post_query.add(post_query_visibility_query, Q.AND)
 
-        post_query_visibility_query.add(Q(community__type=Community.COMMUNITY_TYPE_PUBLIC, is_closed=False), Q.OR)
+        # Check its not closed or we are community staff
+        post_query_closed_query = Q(is_closed=False)
 
-        post_query.add(post_query_visibility_query, Q.AND)
+        post_query_closed_staff_query = Q(community__memberships__user__id=self.pk,
+                                          community__memberships__is_administrator=True) | Q(
+            community__memberships__user__id=self.pk,
+            community__memberships__is_moderator=True)
+
+        post_query_closed_query.add(post_query_closed_staff_query, Q.OR)
```

Comment:
here we need to check `if_staff` and `is_closed=True` .. this seems to check `is_post_open` OR `is_staff` ..

Reply:
Ehh the tests pass (?)

Reply:
yea thats what im wondering the tests are pretty extensive, 🤔

Reply:
All I did was take your queries and simplify them, not really sure about the logic that should or not happen here.

This checks for the same things your other queries did 🤷‍♂️

Reply:
Yep, ur right, i maybe missed it while looking at the smaller queries. But the bigger picture adds up 👍